libcanard/canard.c

```diff
@@ -326,18 +332,39 @@ CANARD_PRIVATE struct CanardTxQueueItem* txAllocateQueueItem(struct CanardTxQueu
     return out;
 }
 
+#define CONTAINER_OF(type, ptr, member) ((type*) (((ptr) == NULL) ? NULL : ((char*) (ptr) - offsetof(type, member))))
```

Comment:
I think this should be near the top. Also, there is risk of this expression accidentally casting away a const qualifier on the pointer. Maybe we should have const and mutable editions of this macro, where the const one would use `const type*` and `const char*`?

Reply:
>Maybe we should have const and mutable editions of this macro, where the const one would use const type* and const char*?

Good point - I will move it to the top, and make two versions: `CONTAINER_OF` & `CONST_CONTAINER_OF`

Reply:
Usually we assume immutability by default, like with the payload containers, for example.

Reply:
Do you mean to have `CONTAINER_OF` (const version) and `MUTABLE_CONTAINER_OF ` (mutable version) ?